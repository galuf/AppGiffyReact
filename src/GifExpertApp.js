import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaulCategories = [] }) => {
  //const [categories, setCategories] = useState(["Pandas"]);
  const [categories, setCategories] = useState(defaulCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((categoria) => (
          <GifGrid key={categoria} category={categoria} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
