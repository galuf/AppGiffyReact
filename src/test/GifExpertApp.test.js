import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Probando componente <GidExpertApp />", () => {
  let wraper = shallow(<GifExpertApp />);

  test("debe mostrarse correctamente", () => {
    expect(wraper).toMatchSnapshot();
  });

  test("debe tener una lista de categorias", () => {
    const categories = ["One Punch", "pandas"];

    const wraper = shallow(<GifExpertApp defaulCategories={categories} />);

    expect(wraper).toMatchSnapshot();
    expect(wraper.find("GifGrid").length).toBe(categories.length);
  });
});
