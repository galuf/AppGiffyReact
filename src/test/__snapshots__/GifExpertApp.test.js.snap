// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GidExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probando componente <GidExpertApp /> debe tener una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="pandas"
      key="pandas"
    />
  </ol>
</Fragment>
`;
